index.js

```diff
@@ -21,33 +21,46 @@ const port = process.env.PORT;
 async function createApp() {
   await connectToDb();
 
+  app.use(
+    expressWinston.logger({
+      transports: [new winston.transports.Console()],
+      format: winston.format.combine(
+        winston.format.colorize(),
+        winston.format.json()
+      ),
+    })
+  );
+
   const app = express();
   // Middleware: parse the body to json
   app.use(bodyParser.json());
   // Api documentation
   app.use("/api-docs", swaggerUi.serve, swaggerUi.setup(swaggerDocument));
 
   app.use(
-    expressWinston.logger({
+    expressWinston.errorLogger({
       transports: [new winston.transports.Console()],
       format: winston.format.combine(
         winston.format.colorize(),
         winston.format.json()
       ),
     })
   );
+
   // Retrieve all orders
   app.get("/", async (req, res) => {
     try {
       const orders = await Order.find();
       res.json(orders);
     } catch (err) {
-      res.status(500).json({ message: err.message });
+      res.status(500).json({ message: "Internal server error" });
     }
   });
   // Retrieve an order with the orderId
   app.get("/:orderId", async (req, res) => {
     const { orderId } = req.params;
+
+    if (!orderId) res.status(400).end();
```

Comment:
Can also send and error message with `res.json({message: "Order not found"})`

Reply:
Done